composer.json

```diff
@@ -34,6 +34,7 @@
         "symfony/console": "^2.8 || ^3.2 || ^4.0",
         "symfony/dependency-injection": "^2.8 || ^3.2 || ^4.0",
         "symfony/event-dispatcher": "^2.8 || ^3.2 || ^4.0",
```

Comment:
Wo we even need this dependency?

Reply:
Yes, there are several usages of the classes provided by this dependency:
https://github.com/sonata-project/SonataAdminBundle/blob/a2524e7a6f03de2f8b8e84203088f6e57b209a30/src/Event/AdminEventExtension.php#L23

Reply:
I mean the `symfony/event-dispatcher` dependency. Do we still need this or can this be replaced with the contract dependency?

Reply:
contracts just contain interface and event object, so I think we need both

Reply:
I think we only need the event symfony/event-dispatcher as it provides the contracts, uses them and have them as dependency. But maybe we need to bump the minimum version

Reply:
If we reference the a package in our code, then we must reference it explicitely, and not rely on package A or B to provide it. That rule can be checked with https://github.com/maglnet/ComposerRequireChecker/

Reply:
But I think we will need to bump this dependency though: does the 2.8 version class extend a class from the contract package? If not, we have a BC-break.

Reply:
Why you consider it as a BC break? The interface from contracts is not adding new methods, and we aren't using declarations against the contract at all, but checking against `instanceof` to detect if the method signature for the `dispatch()` method must be changed.

Reply:
Uuuh right I'm re-reading what I wrote this morning and it no longer makes sense to me :sweat_smile: forget what I said